### PR TITLE
more validations on app payload

### PR DESCRIPTION
**Pull request by bbklab**
http://jira.dataman-inc.com/browse/BORSP-1115

### CODE REVIEW

File: src/manager/framework/state/app.go [hunk 1]
```diff
@@ -401,17 +402,28 @@ func validateAndFormatVersion(version *types.Version) error {
...
 
-	if version.Instances == 0 {
+	if version.Instances <= 0 {
 		return errors.New("invalid instances: instances must be specified and should greater than 0")
 	}
 
+	if version.CPUs < 0.01 {
+		return errors.New("cpu should >= 0.01")
+	}
+	if version.Mem < 16 {
```
Comment (cmingxu) [inline src/manager/framework/state/app.go]: how about make this to 5M, as 16M is a big number for small process.

File: src/manager/framework/state/app.go [hunk 2]
```diff
@@ -401,17 +402,28 @@ func validateAndFormatVersion(version *types.Version) error {
...
+		return errors.New("mem should >= 16")
```
Comment (cmingxu) [inline src/manager/framework/state/app.go]: em should >= 16m

File: src/manager/framework/state/app.go [hunk 3]
```diff
@@ -401,17 +402,28 @@ func validateAndFormatVersion(version *types.Version) error {
 		return errors.New("swan only support mesos docker containerization, no container found")
 	}
 
-	if version.AppName == "" {
-		return errors.New("invalid appName: appName was empty")
+	if err := version.Container.Docker.Valid(); err != nil {
+		return err
+	}
+
+	if n := len(version.AppName); n == 0 || n > 46 {
```
Comment (cmingxu) [inline src/manager/framework/state/app.go]: Not quite sure why 46 is the maximum `AppName` size.

File: src/types/version.go [hunk 4]
```diff
@@ -39,6 +41,13 @@ type Docker struct {
 	Privileged     bool           `json:"privileged,omitempty"`
 }
 
+func (d *Docker) Valid() error {
```
Comment (cmingxu) [inline src/types/version.go]: I don't this is the good way to valid version as 1) version struct under type package are intend to be imported by other project; 2) will make docker validation into multiple occurrences.

Comment (bbklab) [inline src/types/version.go]: 比较理想应该是用户页面传入对象都有一个exported的Valid()方法, 先挪到validateAndFormatVersion中了, 保持和之前一致.

**Comment by bbklab** [comment]
AppName长度最大值, 内存分配最小值,  具体约束数字谁来确定 . @xychu

**Comment by xychu** [comment]
这些值都由 swan 来定，如果不想做限制，就没有限制。 @bbklab

**Comment by bbklab** [comment]
暂定内存5m, cpu 0.01, 长度和页面js限制保持一致48

**Comment by cmingxu** [comment]
LGTM

**Comment by pwzgorilla** [comment]
👍

**Pull request merged**